stdlib/universe/map_vectorize_float_test.flux

```diff
@@ -0,0 +1,71 @@
+package universe_test
+
+
+import "array"
+import "testing"
+
+testcase vec_with_float {
+    want =
+        array.from(
+            rows: [
+                {
+                    // float
+                    f1: 123.0,
+                    f2: 123.0,
+                    // int
+                    i1: 456,
+                    i2: 456.0,
+                    // uint
+                    u1: uint(v: 789),
+                    u2: 789.0,
+                    // string
+                    s1: "1011.12",
+                    s2: 1011.12,
+                    // bool false
+                    b1F: false,
+                    b2F: 0.0,
+                    // bool true
+                    b1T: true,
+                    b2T: 1.0,
+                },
+            ],
+        )
+
+    got =
+        array.from(
+            rows: [
+                {
+                    f1: 123.0,
+                    i1: 456,
+                    u1: uint(v: 789),
+                    s1: "1011.12",
+                    b1F: false,
+                    b1T: true,
+                },
+            ],
+        )
+            |> map(
+                fn: (r) =>
+                    ({r with
+                        f2: float(v: r.f1),
+                        i2: float(v: r.i1),
+                        u2: float(v: r.u1),
+                        s2: float(v: r.s1),
+                        b2F: float(v: r.b1F),
+                        b2T: float(v: r.b1T),
+                    }),
+            )
+
+    testing.diff(want, got)
+}
+
+// FIXME: add bool literals to this once they are enabled in #4997
+testcase vec_with_float_const {
+    want = array.from(rows: [{x: 1, f: 123.456, i: 123.0}])
+
+    got =
+        array.from(rows: [{x: 1}])
+            |> map(fn: (r) => ({r with f: float(v: 123.456), i: float(v: 123)}))
+
+    testing.diff(want, got)
+}
```

Comment:
Does it make sense to add a call to the stdlib's `toFloat` function in here?

It would be nice if we moved the vectorization logic into a planner rule so we could assert on the rule enabling vectorization.

Reply:
I was just looking at queries in lovelace to see what would be needed and I was _literally just thinking about this_. For lovelace I'm planning to have 2 queries, one with a hand-written cast (like you see here) and a 2nd relying on `toFloat()` instead.

I will expect to see the same perf increase for both when we flip the feature flag.

I'm happy to add a test here using `toFloat()` and verify we hit the vectorized path using the debugger before this merges.

Reply:
So, I wrote a test to compare a plain map against the tables produced by `toFloat()` and it turns out, the `toFloat()` call is _not_ vectorizing while the `map()` _is_. This is unexpected.

I can investigate further, but I wonder if this should hold up the PR from merging. I'd like to get a better understanding of why these act different just to strengthen my mental model.

Reply:
I've opened https://github.com/influxdata/flux/issues/5102 to investigate this further, but it looks like we might just not be vectorizing anything that isn't defined in the main package right now.

Redefining `toFloat` in `main` like so, then using it, showed it hits the vectorized code path.

```
toFloat2 = (tables=<-) => tables |> map(fn: (r) => ({r with _value: float(v:r._value)}))
```

I'm guessing there are could be other functions in stdlib that could be vectorized but aren't.